src/components/Dropdown/Dropdown.tsx

```diff
@@ -76,14 +84,26 @@ export const Dropdown = ({
 
   const classes = useMemo(() => classNames(styles.dropdownWrapper, uniqueClassName), [uniqueClassName])
 
+  const onOpenChangeInternal = useCallback(
+    (open: boolean, info?: {source: 'trigger'| 'menu'}) => {
+      if (!onOpenChange) {
+        return
+      }
+      onOpenChange(open, info ? { source: antdSourceMap[info.source] } : undefined)
+    },
+    [onOpenChange]
+  )
+
   return (
     <AntdDropdown
       autoFocus={autoFocus}
       disabled={isDisabled}
       dropdownRender={dropdownRender}
+      getPopupContainer={getPopupContainer}
       menu={menu}
       overlayClassName={classes}
       trigger={triggers}
+      onOpenChange={onOpenChange ? onOpenChangeInternal : undefined}
```

Comment:
since the `onOpenChange` is already verified at lin 89, you can simply leave here our internal callback set

```suggestion
      onOpenChange={onOpenChangeInternal}
```